tests/test_preview.py

```diff
@@ -250,3 +257,116 @@ def test_that_logo_files_exist(logo):
 def test_that_logos_must_have_at_least_one_file(partially_initialised_class):
     with pytest.raises(TypeError):
         partially_initialised_class()
+
+
+@pytest.mark.parametrize('filetype, sentence_count, page_number, expected_response_code', [
+    ('png', 10, 1, 200),
+    ('png', 10, 2, 400),
+    ('png', 50, 2, 200),
+    ('png', 50, 3, 400),
+])
+def test_get_image_by_page_cached(
+        app,
+        mocker,
+        client,
+        auth_header,
+        filetype,
+        sentence_count,
+        page_number,
+        expected_response_code
+):
+    with set_config(app, 'REDIS_ENABLED', True):
+
+        notification_data = json.dumps({
+            'letter_contact_block': '123',
+            'template': {
+                'id': 1,
+                'subject': 'letter subject',
+                'content': (
+                    'All work and no play makes Jack a dull boy. ' * sentence_count
+                ),
+                'version': 1
+            },
+            'values': {},
+            'dvla_org_id': '001',
+        })
+
+        mocked_redis_get = mocker.patch('app.preview.current_app.redis_store.get', return_value=None)
+        mocked_redis_set = mocker.patch('app.preview.current_app.redis_store.set')
+
+        response = client.post(
+            url_for('preview_blueprint.view_letter_template', filetype=filetype, page=page_number),
+            data=notification_data,
+            headers={
+                'Content-type': 'application/json',
+                **auth_header
+            }
+        )
+
+        assert response.status_code == expected_response_code
+
+        unique_name_dict = {
+            'template_id': 1,
+            'version': 1,
+            'dvla_org_id': '001',
+            'letter_contact_block': '123',
+            'values': None
+        }
+
+        assert mocked_redis_get.call_count == 1
+        mocked_redis_get.assert_called_once_with(sorted(unique_name_dict.items()))
+        assert mocked_redis_set.call_count == 1
+
+        mocked_redis_get = mocker.patch('app.preview.current_app.redis_store.get', return_value=response.data)
```

Comment:
these should be separate unit tests

Reply:
No it should have been testing that a cached pdf is used, but it wasn't.
I've updated the parameters and the code to o that and changed the name of the method to make it clearer.